bert/src/model.rs

```diff
@@ -144,209 +68,121 @@ impl Runtime {
+        let input = &model.runtime.inputs[1];
         assert_eq!(input.dimensions, [None, None]);
         assert_eq!(input.input_type, TensorElementDataType::Int64);
-        let input = &session.inputs[2];
+        let input = &model.runtime.inputs[2];
         assert_eq!(input.dimensions, [None, None]);
         assert_eq!(input.input_type, TensorElementDataType::Int64);
 
-        assert_eq!(session.outputs.len(), 2);
-        let output = &session.outputs[0];
+        assert_eq!(model.runtime.outputs.len(), 2);
+        let output = &model.runtime.outputs[0];
         assert_eq!(output.dimensions, [None, None, None]);
         assert_eq!(output.output_type, TensorElementDataType::Float32);
-        let output = &session.outputs[1];
-        assert_eq!(output.dimensions, [None, Some(embedding_size)]);
+        let output = &model.runtime.outputs[1];
+        assert_eq!(output.dimensions, [None, Some(128)]);
         assert_eq!(output.output_type, TensorElementDataType::Float32);
     }
 
     #[test]
-    fn test_predict_tract() {
-        let shape = (1, 64);
-        let config = Config::new(smbert_mocked().unwrap())
-            .unwrap()
-            .with_token_size(shape.1)
-            .unwrap()
-            .with_runtime(config::Runtime::Tract);
-        let model = Model::new(&config).unwrap();
-
-        let encoding = Encoding {
-            token_ids: Array2::from_elem(shape, 0),
-            attention_mask: Array2::from_elem(shape, 1),
-            type_ids: Some(Array2::from_elem(shape, 0)),
-        };
-        let prediction = model.predict(encoding).unwrap();
-        assert_eq!(model.token_size, shape.1);
-        assert_eq!(prediction.shape(), [shape.0, shape.1, model.embedding_size]);
-    }
-
-    #[test]
-    fn test_predict_ort() {
-        let shape = (1, 64);
-        let config = Config::new(smbert_mocked().unwrap())
-            .unwrap()
-            .with_token_size(shape.1)
+    fn test_embed() {
+        let token_size = 64;
+        let config = Config::new(smbert_mocked().unwrap(), ort().unwrap())
             .unwrap()
-            .with_runtime(config::Runtime::Ort(ort().unwrap()));
+            .with_token_size(token_size)
+            .unwrap();
         let model = Model::new(&config).unwrap();
 
-        let encoding = Encoding {
-            token_ids: Array2::from_elem(shape, 0),
-            attention_mask: Array2::from_elem(shape, 1),
-            type_ids: Some(Array2::from_elem(shape, 0)),
-        };
-        let prediction = model.predict(encoding).unwrap();
-        assert_eq!(model.token_size, shape.1);
-        assert_eq!(prediction.shape(), [shape.0, shape.1, model.embedding_size]);
+        let encoding = Encoding::new(
+            vec![0; token_size],
+            vec![0; token_size],
+            Vec::new(),
+            Vec::new(),
+            Vec::new(),
+            Vec::new(),
+            vec![1; token_size],
+            Vec::new(),
+            HashMap::new(),
+        );
```

Comment:
:art:  This constructor falls cleanly into the "never write code like that" area, but it's also outside of our code, so do we keep it like that?

Reply:
i know it's not nice 😅 it's the shortest workaround i found to set up the test, and since we know that the unset parts are not used, it should be ok to keep it. wdyt?